backend/src/main/java/com/github/johannesthorbergsson/backend/orders/OrderService.java

```diff
@@ -0,0 +1,21 @@
+package com.github.johannesthorbergsson.backend.orders;
+
+import com.github.johannesthorbergsson.backend.id.IdService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.security.Principal;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class OrderService {
+    private final OrderRepository orderRepository;
+    private final IdService idService;
+
+    public List<ServiceOrder> getAllOrders(Principal principal){
+        return orderRepository.findAll().stream()
+                .filter(order -> order.username().equals(principal.getName())).toList();
```

Comment:
I suggest building a custom method in the repo. It should be possible to write a method signature like `findByUsername(string username)`. That eliminates the possibility that you make a mistake in the stream filter :)

Reply:
Smart